internal/services/appservice/windows_function_app_resource.go

```diff
@@ -536,6 +547,22 @@ func (r WindowsFunctionAppResource) Read() sdk.ResourceFunc {
 				return fmt.Errorf("reading Windows %s: %+v", id, err)
 			}
 
+			var planSku *string
+			if functionApp.ServerFarmID != nil {
+				servicePlanId, err := parse.ServicePlanID(*functionApp.ServerFarmID)
+				if err != nil {
+					return err
+				}
+				servicePlan, err := servicePlanClient.Get(ctx, servicePlanId.ResourceGroup, servicePlanId.ServerfarmName)
+				if err != nil {
+					return fmt.Errorf("reading %s: %+v", servicePlanId, err)
+				}
+				if sku := servicePlan.Sku; sku != nil && sku.Name != nil {
+					planSku = sku.Name
+				}
+			}
+			sendContentSettings := helpers.PlanIsElastic(planSku)
+
```

Comment:
As above, this functionality shouldn't be present in the Read.

Reply:
We need to get the service plan id to determine if the function app is elastic or not, then to check if the website_content_share property.